src/skmultiflow/data/base_stream.py

```diff
@@ -236,7 +237,10 @@ def is_restartable(self):
 
     def restart(self):
         """  Restart the stream. """
-        self.prepare_for_use()
+        self.current_sample_x = None
+        self.current_sample_y = None
+        self.sample_idx = 0
+        self._prepare_for_use()
```

Comment:
Would not this call raise a `NotImplementedError`?

Reply:
the `Stream` class is an abstract class so it should not be used as-is. Additionally, `_prepare_for_use` is an `@abstractmethod` so it must be implemented by all extensions of the `Stream` class.

Reply:
You are right. Sorry, I got confused. The change is correct!